doc/neo-tree.txt

```diff
@@ -1438,11 +1438,19 @@ This is fired inside a vim.schedule.
 >lua
     {
       event = "neo_tree_popup_input_ready",
-      ---@param input NuiInput
-      handler = function(input)
+      handler = function()
         -- enter input popup with normal mode by default.
         vim.cmd("stopinsert")
```

Comment:
This example should probably contain an override for the `<esc>` mapping too, shouldn't it? I don't think that anyone that uses normal mode for inputs would want `<esc>` to close the input from insert mode.

Reply:
That's in the second event handler. I thought you said we should split the examples apart.

Reply:
https://github.com/nvim-neo-tree/neo-tree.nvim/pull/1398#issuecomment-1999472063

Reply:
Yes, theree should be separate examples, but all of them requires changing the default `<esc>` mapping in a different way. Here are some examples:

1. Start in normal mode, change escape so that it only acts as the standard "back to normal mode" action
2. Start in insert mode, but allow escape to exit insert mode, and the next escape from normal mode to close the popup.
3. Start in insert mode, but allow escape to exit insert mode, use q to close the popup

It goes on and on, which is why we are going with the one standard "plus make your own." The issue is that in every scenario where you start in normal mode, you would want to change the default `<esc>` mapping because in some way so that it is possible to leave insert mode and go back to normal mode.

Reply:
If you have something like `nnoremap jk <Esc>`, you don't need `<Esc>` remapped to go to normal mode. In fact this is what I do; `jk` to go to normal mode, `<Esc>` to quit from within insert mode and ngl it's pretty convenient.

> The issue is that in every scenario where you start in normal mode, you would want to change the default <esc> mapping in some way so that it is possible to leave insert mode and go back to normal mode.

So I don't agree this is for every case.

Reply:
OK, fair enough. I guess that goes back to the point of why we are doing this in the first place, just too many variations.

Reply:
> just too many variations.

99% of the issues we have comes down to this for real lol.

Reply:
Very true.